src/main/java/org/springframework/data/elasticsearch/client/RestClients.java

```diff
@@ -105,15 +106,14 @@ public static ElasticsearchRestClient create(ClientConfiguration clientConfigura
 			Duration connectTimeout = clientConfiguration.getConnectTimeout();
 
 			if (!connectTimeout.isNegative()) {
-
 				requestConfigBuilder.setConnectTimeout(Math.toIntExact(connectTimeout.toMillis()));
-				requestConfigBuilder.setConnectionRequestTimeout(Math.toIntExact(connectTimeout.toMillis()));
 			}
 
 			Duration timeout = clientConfiguration.getSocketTimeout();
 
 			if (!timeout.isNegative()) {
 				requestConfigBuilder.setSocketTimeout(Math.toIntExact(timeout.toMillis()));
+				requestConfigBuilder.setConnectionRequestTimeout(Math.toIntExact(connectTimeout.toMillis()));
```

Comment:
just at a second look I saw that you moved the line into the _if_ for the socketTimeout, but still using the configured value for the connectTimeout. Shouldn't do PRs in the morning before work. I'll fix that.